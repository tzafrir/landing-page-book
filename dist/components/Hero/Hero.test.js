'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _Hero = require('./Hero');

var _Hero2 = _interopRequireDefault(_Hero);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('Hero', function () {
  var props = void 0;
  var mounted = void 0;
  var hero = function hero() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_Hero2.default, props));
    }
    return mounted;
  };
  describe('Hero', function () {
    beforeEach(function () {
      props = {
        isFixed: true,
        isCentered: true,
        particles: true,
        opacity: 0,
        parallaxOffset: 0,
        image: '',
        overlayColor: '#fff',
        header: 'AAAAAAAAAAAAAAA',
        subHeader: 'BBBBBBBBBB',
        subHeaderPosition: 'bottom',
        minHeight: '100vh',
        backgroundColor: 'transparent',
        buttons: [],
        particlesParams: {},
        children: null
      };
      mounted = undefined;
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_Hero2.default, props))).toMatchSnapshot();
    });

    it('always renders a div', function () {
      var divs = hero().find('div');

      expect(divs.length).toBeGreaterThan(0);
    });
  });
});