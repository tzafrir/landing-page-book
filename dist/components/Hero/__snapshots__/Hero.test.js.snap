// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Hero should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hero
    backgroundColor="transparent"
    buttons={Array []}
    header="AAAAAAAAAAAAAAA"
    image=""
    imgFilter={null}
    isCentered={true}
    isFixed={true}
    minHeight="100vh"
    opacity={0}
    overlayColor="#fff"
    parallaxOffset={0}
    particles={true}
    particlesParams={Object {}}
    particlesSugar={null}
    style={null}
    subHeader="BBBBBBBBBB"
    subHeaderPosition="bottom"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <LazyHero
        className={undefined}
        color="#fff"
        imageSrc=""
        isCentered={true}
        isFixed={true}
        minHeight="100vh"
        opacity={0}
        parallaxOffset={0}
        style={null}
        transitionDuration={600}
        transitionTimingFunction="ease-in-out"
      >
        <div
          className="Hero__Particles"
        >
          <t
            height="100%"
            params={Object {}}
            style={Object {}}
            width="100%"
          />
        </div>
        <div
          className="Hero__Container d-flex align-content-center"
        >
          <Container
            className=""
            fluid={false}
          >
            <Header
              borderBottom={false}
              borderTop={false}
              className="Hero__Header mb-5"
              color=""
              icon=""
              textAlign="center"
              type="h1"
            >
              AAAAAAAAAAAAAAA
            </Header>
            <Header
              borderBottom={false}
              borderTop={false}
              className="Hero__SubHeader mb-5"
              color=""
              icon=""
              textAlign="center"
              type="h2"
            >
              BBBBBBBBBB
            </Header>
          </Container>
        </div>
      </LazyHero>,
      "className": "Hero Hero_1541105775050_04648036763981598",
      "style": Object {
        "backgroundColor": "transparent",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="Hero__Particles"
          >
            <t
              height="100%"
              params={Object {}}
              style={Object {}}
              width="100%"
            />
          </div>,
          <div
            className="Hero__Container d-flex align-content-center"
          >
            <Container
              className=""
              fluid={false}
            >
              <Header
                borderBottom={false}
                borderTop={false}
                className="Hero__Header mb-5"
                color=""
                icon=""
                textAlign="center"
                type="h1"
              >
                AAAAAAAAAAAAAAA
              </Header>
              <Header
                borderBottom={false}
                borderTop={false}
                className="Hero__SubHeader mb-5"
                color=""
                icon=""
                textAlign="center"
                type="h2"
              >
                BBBBBBBBBB
              </Header>
            </Container>
          </div>,
          null,
        ],
        "className": undefined,
        "color": "#fff",
        "imageSrc": "",
        "isCentered": true,
        "isFixed": true,
        "minHeight": "100vh",
        "opacity": 0,
        "parallaxOffset": 0,
        "style": null,
        "transitionDuration": 600,
        "transitionTimingFunction": "ease-in-out",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <t
              height="100%"
              params={Object {}}
              style={Object {}}
              width="100%"
            />,
            "className": "Hero__Particles",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "height": "100%",
              "params": Object {},
              "style": Object {},
              "width": "100%",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Container
                className=""
                fluid={false}
              >
                <Header
                  borderBottom={false}
                  borderTop={false}
                  className="Hero__Header mb-5"
                  color=""
                  icon=""
                  textAlign="center"
                  type="h1"
                >
                  AAAAAAAAAAAAAAA
                </Header>
                <Header
                  borderBottom={false}
                  borderTop={false}
                  className="Hero__SubHeader mb-5"
                  color=""
                  icon=""
                  textAlign="center"
                  type="h2"
                >
                  BBBBBBBBBB
                </Header>
              </Container>,
              null,
            ],
            "className": "Hero__Container d-flex align-content-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  false,
                  <Header
                    borderBottom={false}
                    borderTop={false}
                    className="Hero__Header mb-5"
                    color=""
                    icon=""
                    textAlign="center"
                    type="h1"
                  >
                    AAAAAAAAAAAAAAA
                  </Header>,
                  <Header
                    borderBottom={false}
                    borderTop={false}
                    className="Hero__SubHeader mb-5"
                    color=""
                    icon=""
                    textAlign="center"
                    type="h2"
                  >
                    BBBBBBBBBB
                  </Header>,
                  Array [],
                ],
                "className": "",
                "fluid": false,
              },
              "ref": null,
              "rendered": Array [
                false,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "borderBottom": false,
                    "borderTop": false,
                    "children": "AAAAAAAAAAAAAAA",
                    "className": "Hero__Header mb-5",
                    "color": "",
                    "icon": "",
                    "textAlign": "center",
                    "type": "h1",
                  },
                  "ref": null,
                  "rendered": "AAAAAAAAAAAAAAA",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "borderBottom": false,
                    "borderTop": false,
                    "children": "BBBBBBBBBB",
                    "className": "Hero__SubHeader mb-5",
                    "color": "",
                    "icon": "",
                    "textAlign": "center",
                    "type": "h2",
                  },
                  "ref": null,
                  "rendered": "BBBBBBBBBB",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            null,
          ],
          "type": "div",
        },
        null,
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <LazyHero
          className={undefined}
          color="#fff"
          imageSrc=""
          isCentered={true}
          isFixed={true}
          minHeight="100vh"
          opacity={0}
          parallaxOffset={0}
          style={null}
          transitionDuration={600}
          transitionTimingFunction="ease-in-out"
        >
          <div
            className="Hero__Particles"
          >
            <t
              height="100%"
              params={Object {}}
              style={Object {}}
              width="100%"
            />
          </div>
          <div
            className="Hero__Container d-flex align-content-center"
          >
            <Container
              className=""
              fluid={false}
            >
              <Header
                borderBottom={false}
                borderTop={false}
                className="Hero__Header mb-5"
                color=""
                icon=""
                textAlign="center"
                type="h1"
              >
                AAAAAAAAAAAAAAA
              </Header>
              <Header
                borderBottom={false}
                borderTop={false}
                className="Hero__SubHeader mb-5"
                color=""
                icon=""
                textAlign="center"
                type="h2"
              >
                BBBBBBBBBB
              </Header>
            </Container>
          </div>
        </LazyHero>,
        "className": "Hero Hero_1541105775050_04648036763981598",
        "style": Object {
          "backgroundColor": "transparent",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="Hero__Particles"
            >
              <t
                height="100%"
                params={Object {}}
                style={Object {}}
                width="100%"
              />
            </div>,
            <div
              className="Hero__Container d-flex align-content-center"
            >
              <Container
                className=""
                fluid={false}
              >
                <Header
                  borderBottom={false}
                  borderTop={false}
                  className="Hero__Header mb-5"
                  color=""
                  icon=""
                  textAlign="center"
                  type="h1"
                >
                  AAAAAAAAAAAAAAA
                </Header>
                <Header
                  borderBottom={false}
                  borderTop={false}
                  className="Hero__SubHeader mb-5"
                  color=""
                  icon=""
                  textAlign="center"
                  type="h2"
                >
                  BBBBBBBBBB
                </Header>
              </Container>
            </div>,
            null,
          ],
          "className": undefined,
          "color": "#fff",
          "imageSrc": "",
          "isCentered": true,
          "isFixed": true,
          "minHeight": "100vh",
          "opacity": 0,
          "parallaxOffset": 0,
          "style": null,
          "transitionDuration": 600,
          "transitionTimingFunction": "ease-in-out",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <t
                height="100%"
                params={Object {}}
                style={Object {}}
                width="100%"
              />,
              "className": "Hero__Particles",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "height": "100%",
                "params": Object {},
                "style": Object {},
                "width": "100%",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Container
                  className=""
                  fluid={false}
                >
                  <Header
                    borderBottom={false}
                    borderTop={false}
                    className="Hero__Header mb-5"
                    color=""
                    icon=""
                    textAlign="center"
                    type="h1"
                  >
                    AAAAAAAAAAAAAAA
                  </Header>
                  <Header
                    borderBottom={false}
                    borderTop={false}
                    className="Hero__SubHeader mb-5"
                    color=""
                    icon=""
                    textAlign="center"
                    type="h2"
                  >
                    BBBBBBBBBB
                  </Header>
                </Container>,
                null,
              ],
              "className": "Hero__Container d-flex align-content-center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    false,
                    <Header
                      borderBottom={false}
                      borderTop={false}
                      className="Hero__Header mb-5"
                      color=""
                      icon=""
                      textAlign="center"
                      type="h1"
                    >
                      AAAAAAAAAAAAAAA
                    </Header>,
                    <Header
                      borderBottom={false}
                      borderTop={false}
                      className="Hero__SubHeader mb-5"
                      color=""
                      icon=""
                      textAlign="center"
                      type="h2"
                    >
                      BBBBBBBBBB
                    </Header>,
                    Array [],
                  ],
                  "className": "",
                  "fluid": false,
                },
                "ref": null,
                "rendered": Array [
                  false,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "borderBottom": false,
                      "borderTop": false,
                      "children": "AAAAAAAAAAAAAAA",
                      "className": "Hero__Header mb-5",
                      "color": "",
                      "icon": "",
                      "textAlign": "center",
                      "type": "h1",
                    },
                    "ref": null,
                    "rendered": "AAAAAAAAAAAAAAA",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "borderBottom": false,
                      "borderTop": false,
                      "children": "BBBBBBBBBB",
                      "className": "Hero__SubHeader mb-5",
                      "color": "",
                      "icon": "",
                      "textAlign": "center",
                      "type": "h2",
                    },
                    "ref": null,
                    "rendered": "BBBBBBBBBB",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              null,
            ],
            "type": "div",
          },
          null,
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
